pkg/pipeline/encode/opentelemetry/encode_otlptrace.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2023 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package opentelemetry
+
+import (
+	"context"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/operational"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/encode"
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	flpTracerName     = "flp_tracer"
+	flpEncodeSpanName = "flp_encode"
+)
+
+type EncodeOtlpTrace struct {
+	cfg    api.EncodeOtlpTraces
+	ctx    context.Context
+	res    *resource.Resource
+	tp     *sdktrace.TracerProvider
+	tracer trace.Tracer
+}
+
+// Encode encodes a metric to be exported
+func (e *EncodeOtlpTrace) Encode(metricRecord config.GenericMap) {
+	log.Tracef("entering EncodeOtlpTrace. entry = %v", metricRecord)
+	_, span := e.tracer.Start(e.ctx, flpEncodeSpanName)
+	defer span.End()
+	attributes := obtainAttributesFromEntry(metricRecord)
+	span.SetAttributes(*attributes...)
+}
```

Comment:
I am not sure what is the goal here with traces, but I don't think we're doing something useful here.
Normally, traces are provided by the workloads that are being traced, for instance when a workload A sends a request to service S, A would generate a span saying "I am sending a request to S".

I believe what we could possibly do, as an external observer, is to generate spans on behalf of the workloads & services as long as we see their IPs flowing. I'm not 100% sure how useful it is, but that's the best I can see. But it's not really what this is doing here: it creates spans for FLP without providing any real information about the network connections being observed in otel format.